pynio/block.py

```diff
@@ -40,3 +40,11 @@ def name(self, value):
     @property
     def type(self):
         return self._type
+
+    def delete(self):
+        '''Delete self from instance and services'''
+        self._instance.blocks.remove(self._name)
+        for s in self._instance.services.values():
+            s._remove(self)
```

Comment:
I think you should do that a different way. I personally think the system of swapping out blocks by name is a terrible convetion and shouldn't be upheld in an api.

If you want to replace a block with a different block, do it explicitly -- not implicitly by using the sane name. If that is something we want, we should implement a `replace` function.